### ISSUE TITLE
Consider the usage of staticcall for read only cross-contract calls

### ISSUE THREAD

**Issue by sangier** [cosmos/solidity-ibc-eureka#48]
Staticcall can be useful for functions where only a return value is desired and state changes are unwanted. It add an extra layer of safety since it will revert if the contract called causes a state change and can save some gas. 

Food for thoughts:
- https://github.com/ethereum/EIPs/blob/master/EIPS/eip-214.md 
- https://www.rareskills.io/post/solidity-staticcall#:~:text=Staticcall%20is%20like%20a%20regular,in%20solidity%20function%20named%20staticcall.

1. More investigation is needed.
2. The snippets of code where it can be applied should be identified.

**Comment by gjermundgaraba** [comment]
I don't think we need to worry about this right now, but we can reconsider in the future if needed
